### ISSUE TITLE
Date type takes up a lot of vertical space in grid mode print view

### ISSUE THREAD

**Issue by lognaturel** [enketo/enketo-core#705]
In the online grid view, questions of type date take up more vertical space than expected when compared to other field types but it's very usable:
<img width="232" alt="Screen Shot 2020-04-10 at 3 53 33 PM" src="https://user-images.githubusercontent.com/967540/79028040-74279d80-7b43-11ea-97b4-3d69684e3c82.png">

In print view they take up too much vertical space to be usable for printing forms to be filled out:
<img width="68" alt="Screen Shot 2020-04-10 at 3 55 16 PM" src="https://user-images.githubusercontent.com/967540/79028111-b7820c00-7b43-11ea-9785-55fb6473ccfe.png">

When putting a date next to another question type, the grid is broken in print view:
<img width="481" alt="Screen Shot 2020-04-10 at 4 02 11 PM" src="https://user-images.githubusercontent.com/967540/79028396-b1d8f600-7b44-11ea-92b1-16e138197238.png">

**Comment by MartijnR** [comment]
* [ ] date too much vertical space
* [ ] text too little vertical space (unless tall neigbour is present)
* [ ] dotted line not pushed down to bottom

**Comment by MartijnR** [comment]
@lognaturel I updated enketo-express with a fix for these 3 issue (not released yet). It's deployed on odk.enke.to. If you have time, please have look, and reopen if you still see similar issues. Thanks!

**Comment by lognaturel** [comment]
Brilliant! This is much better, thanks so much. 

There are still a couple cases where the layout decisions aren't the same:
* Some columns don't end up next to each other. I think it has to do with relevance but I'm not entirely sure. <img width="488" alt="Screen Shot 2020-04-14 at 10 55 12 AM" src="https://user-images.githubusercontent.com/967540/79257547-a20c2b00-7e3e-11ea-80be-f33ec088fe69.png">

* One of the select questions ends up with extra spacing. These three are also next to each other in the web view. <img width="480" alt="Screen Shot 2020-04-14 at 10 55 28 AM" src="https://user-images.githubusercontent.com/967540/79257628-c36d1700-7e3e-11ea-8b0a-993183195b70.png">

I'm working on a version of this form that can be publicly shared. I'll update the thread here with it so you can try it out in context.

**Comment by MartijnR** [comment]
Thanks. What you're seeing with cells not ending up on the same row could also have to do with nesting (I'm referring to point 4 of the intro of https://enke.to/grid).

**Comment by lognaturel** [comment]
I did see the note and took it into account. The cells that are ending up on separate rows are only in a single group. See [the form](https://docs.google.com/spreadsheets/d/125HOZjkgA5dg8FQVMCetHWTGHvEwfRx8J0w7h1ofMoM/edit#gid=0). The two screenshots I shared start at `patcourse_symptoms` and `patcourse_icu`.

The three questions are repeated in [the followup form](https://docs.google.com/spreadsheets/d/1v0i_55DDFJb-dbX_-k1Di1iMK0VPQUgeRh_ibDHIAxo/edit#gid=103335148) and in that case those are in a nested group. Those actually show up as expected in the print view. That follow up form also has questions unexpectedly on separate rows.

Another difference I note that makes using the print view challenging is that read-only text fields end up having a dotted line. That means they take up more vertical space:
<img width="78" alt="Screen Shot 2020-04-14 at 1 20 19 PM" src="https://user-images.githubusercontent.com/967540/79271857-5913a100-7e55-11ea-9152-31442ed36fd2.png">

**Comment by MartijnR** [comment]
Thanks! Very helpful!

http://localhost:8005/?xform=COVID1.xml

* [ ] patcourse_symptions
* [ ] follow up form
* [ ] remove dotted line from readonly

**Comment by MartijnR** [comment]
Step closer. Only the pesky dotted line remaining. Difficult to resolve properly. Will probably save until after https://github.com/enketo/enketo-core/issues/391 (which is not planned because it's huge and will require funding).

temporary links to the forms above:
 * https://odk.enke.to/preview?form=https://xlsform.opendatakit.org/downloads/tbrlqopc/Customized%20grid%20WHO%20COVID-19%20Form_%20confirmed%20case%20report%20(part%201).xml
* https://odk.enke.to/preview?form=https://xlsform.opendatakit.org/downloads/reommida/Customized%20grid%20WHO%20COVID-19%20Form_%20confirmed%20case%20report%20(part%202_%20outcome).xml

**Comment by lognaturel** [comment]
That's fantastic, thanks so much, @MartijnR! That should now be usable for my purposes. 🎉🎉🎉

The dotted line isn't a big deal, it's the vertical space that was being taken up that was a problem. Now it looks like it's collapsed. It seems to me that whatever process is collapsing the space should also be usable to remove the dotted line but I'll take your word for it that dynamic readonly evaluation is needed in this case.

Like I said, this now looks great and so no action needed but I wanted to highlight a couple things that would make the print view even more compelling:
- removing the drop shadow on the right
- removing the right gridline line between subgroups and their parent. See the picture below for what I mean. I know you say subgroups aren't supported but they actually look great and I think are very helpful for this form. This is the only issue I can see with them in either web or print view.
<img width="658" alt="Screen Shot 2020-04-15 at 2 04 28 PM" src="https://user-images.githubusercontent.com/967540/79388359-0b189f00-7f22-11ea-9c06-181d2c7bcaf2.png">

If I really could dream, specifying page break points would be amazing. I think that could be a reasonable XForms spec attribute. As you can see in the photo above, I ended up playing with scale to get the page breaks in reasonable places.

But again, the bugs are fixed.

**Comment by MartijnR** [comment]
Thanks @lognaturel! The border has been removed. I have created/highlighted separate issues for the remaining dreams:

* https://github.com/enketo/enketo-core/issues/710
* https://github.com/enketo/enketo-core/issues/561

The line extending at the bottom, is not fixable unfortunately, I think. It's not a subgroup issue. It also occurs with top-level groups. It's a quirky page-break thing, that is likely spec-compliant.
